database/database.go

```diff
@@ -266,3 +268,42 @@ func (s *DatabaseService) GetNumDeliveredPayloads() (uint64, error) {
 	err := s.DB.QueryRow("SELECT COUNT(*) FROM " + TableDeliveredPayload).Scan(&count)
 	return count, err
 }
+
+func (s *DatabaseService) GetBuilderSubmissions(filters GetBuilderSubmissionsFilters) ([]*BuilderBlockSubmissionEntry, error) {
+	arg := map[string]interface{}{
+		"limit":        filters.Limit,
+		"slot":         filters.Slot,
+		"cursor":       filters.Cursor,
+		"block_hash":   filters.BlockHash,
+		"block_number": filters.BlockNumber,
+	}
+
+	tasks := []*BuilderBlockSubmissionEntry{}
+	fields := "id, inserted_at, slot, epoch, builder_pubkey, proposer_pubkey, proposer_fee_recipient, parent_hash, block_hash, block_number, num_tx, value, gas_used, gas_limit"
+
+	whereConds := []string{"sim_success = true"}
+	if filters.Slot > 0 {
+		whereConds = append(whereConds, "slot = :slot")
+	} else if filters.Cursor > 0 {
+		whereConds = append(whereConds, "slot <= :cursor")
```

Comment:
using the slot cursor you can request up to 100 entries at or before any given slot in the past.

Reply:
Because in this endpoint several records are allocated to one slot, the cursor and pagination do not work as expected:
```js
(async () => {
  const res = await fetch('https://builder-relay-sepolia.flashbots.net/relay/v1/data/bidtraces/builder_blocks_received?cursor=509739');
  const json = await res.json()
  const uniqueSlots = [...new Set(json.map(item => item.slot))]
  console.log(json.length, uniqueSlots.length)
})()
// 100 25
```

Reply:
yeah right. this is experimental... it should still be enough the go through the data. 

open to ideas for improvements!